### ISSUE TITLE
The selected item in a Nav component is not read by screen readers as "selected"

### ISSUE THREAD

**Issue by jai-ram** [microsoft/fluentui#22371]
### Environment Information

- **Package version(s)**: 8.64.1
- **Browser and OS versions**: Edge (v 100.0.1185.29, 64 bit), Windows 11 Pro (OS build 22000.556)

### Describe the issue:

When the user goes over the items in a Nav component using a screen reader, he will not be able to determine which Nav component is selected

### Please provide a reproduction of the issue in a codepen:

Taken from your Nav demo, I created this codepen: https://codepen.io/jai-ram/pen/vYpRWEG

#### Actual behavior:

The user cannot distinguish through a screen reader if the item in the Nav is selected or not. "Documents" is selected in the example provider and Narrator reads "2 of 6, level 1, Documents"

#### Expected behavior:

Narrator should announce "2 of 6, level 1, Documents, selected"

### Documentation describing expected behavior

This link goes into the Microsoft intranet, so access to it is required to open the link. 

Microsoft's Accessibility Standards, 4.2.1: Object Information Guide, covers the expected behavior. 

https://microsoft.sharepoint.com/sites/accessibility/SitePages/MAS-4.2.1-Object-Information-Guide.aspx
